### ISSUE TITLE
Pill close button accessibility

### ISSUE THREAD

**Issue by lluisgarciapons** [Sage/carbon#5715]
### Current behaviour

When the pill has _onDelete_ and using Voice-Over, it reads the "X" as _close button_, but it's actually not closing anything.

### Expected behaviour

The "X" should read as _dismiss pill_ or similar

### CodeSandbox or Storybook URL

https://codesandbox.io/s/ancient-frog-jpb2er?file=/src/index.js

### JIRA Ticket (Sage Only)

SBS-48576

### Suggested Solution

use _aria-label_ or _aria-labelledby_ attributes

_No response_

### Carbon Version

111.8.0

### Design Tokens Version

2.45.0

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by tomdavies73** [comment]
@tempertemper what do you think of this? @lluisgarciapons is suggesting the aria-label should be changed to "dismiss" instead of "close" for the Pill component. Do you think this language change is justified, or are we okay sticking with the previous aria-label?

**Comment by tempertemper** [comment]
Closing would be for something that was previously opened, so 'close' doesn't work in this context. 'Dismiss' could work. As could 'remove'. Might be one to get a content designer involved with.

**Comment by tomdavies73** [comment]
FE-5600
